pkg/server/state/state.go

```diff
@@ -143,6 +146,73 @@ func (s *State) GetEnv(ctx context.Context, envSlug string) (libapi.Env, error)
 	return env, nil
 }
 
+func (s *State) GetTaskErrors(ctx context.Context, slug string, envSlug string) (AppCondition, error) {
+	key := appConditionKey(slug, envSlug)
+	if result, ok := s.AppCondition.Get(key); ok {
+		return result, nil
+	}
+
+	result := AppCondition{
+		RefreshedAt: time.Now(),
+	}
+
+	taskConfig, ok := s.TaskConfigs.Get(slug)
+	if !ok {
+		// Not supported locally.
+		kind, _, err := taskConfig.Def.GetKindAndOptions()
+		if err != nil {
+			return AppCondition{}, errors.Wrap(err, "getting task kind")
+		}
+		result.Errors = append(result.Errors, dev_errors.AppError{
+			Level:   dev_errors.LevelError,
+			AppName: taskConfig.Def.GetName(),
+			AppKind: "task",
+			Reason:  fmt.Sprintf("%v task does not support local execution", kind),
+		})
+	} else {
+		mergedResources, err := resources.MergeRemoteResources(ctx, s.RemoteClient, s.DevConfig, pointers.String(envSlug))
+		if err != nil {
+			return AppCondition{}, errors.Wrap(err, "merging local and remote resources")
+		}
+
+		// Check resource attachments.
+		var missingResources []string
+		resourceAttachments, err := taskConfig.Def.GetResourceAttachments()
+		if err != nil {
+			return AppCondition{}, errors.Wrap(err, "getting resource attachments")
+		}
+		for _, ref := range resourceAttachments {
+			if _, ok := resources.LookupResource(mergedResources, ref); !ok {
+				missingResources = append(missingResources, ref)
+			}
+		}
+		if len(missingResources) > 0 {
+			sort.Strings(missingResources)
+			result.Errors = append(result.Errors, dev_errors.AppError{
+				Level:   dev_errors.LevelWarning,
+				AppName: taskConfig.Def.GetName(),
+				AppKind: "task",
+				Reason:  fmt.Sprintf("Attached resource: %v not found in dev config file or remotely.", missingResources),
+			})
+		}
```

Comment:
Unrelated to your change, but the array brackets in this error message have bothered me before. It seems straightforward to create one error per missing resource. What do you think?

```suggestion
		for _, resource := range missingResources {
			result.Errors = append(result.Errors, dev_errors.AppError{
				Level:   dev_errors.LevelWarning,
				AppName: taskConfig.Def.GetName(),
				AppKind: "task",
				Reason:  fmt.Sprintf("Attached resource %q not found in dev config file or remotely.", resource),
			})
		}
```